app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/PostNewClaimController.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers
+
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging
+import uk.gov.hmrc.cdsreimbursementclaim.utils.Logging.LoggerOps
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.Error
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.services.SubmitClaimService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import cats.data.EitherT._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PostNewClaimController @Inject() (eisService: SubmitClaimService)(implicit
+  mcc: MessagesControllerComponents,
+  ec: ExecutionContext
+) extends FrontendController(mcc)
+    with Logging {
+
+  val claim: Action[AnyContent] = Action.async { implicit request =>
+    (request.method match {
+      case "GET"  => rightT[Future, Error](testRequestBody)
+      case "POST" => fromOption[Future](request.body.asJson, Error("Request Body is not Json!"))
+    })
+      .flatMap(eisService.submitClaim)
+      .fold(
+        e => {
+          logger.warn(s"could not submit claim", e)
+          InternalServerError
+        },
+        response => Ok(response)
+      )
+
+  }
+
+  val testRequestBody: JsValue = Json.parse("""{
```

Comment:
We don't know whether this payload is correct or not. This is the payload for the EIS -> CDFPay interaction. The stub service at the moment has no schema to validate against so it will accept anything.

Reply:
Have you cross referenced this with the new draft API spec? It seems to have an initial version of a payload structure